objects.lua

```diff
@@ -242,3 +242,41 @@ formspeclib.register_object('inventory', function(obj, safe_mode)
     return 'list[' .. location .. ';' .. name .. ';' .. x .. ',' .. y .. ';' ..
         w .. ',' .. h .. ';' .. start .. ']' .. extra
 end)
+
+--
+-- A container
+--
+-- Parameters: x, y, children
+--
+-- each element inside the container is offset by the x and y
+--
+formspeclib.register_object('container', function(obj, safe_mode)
+    return {
+        {
+            type = 'formspeclib:container_start',
+            x = formspeclib.escape(obj.x or 0),
+            y = formspeclib.escape(obj.y or 0),
+        },
+        (table.unpack or unpack)(obj),
```

Comment:
Shoudn't this be `obj[1]`?

Reply:
Oh, your syntax is
~~~lua
{
    type = "container",
    x = 1,
    y = 1,
    {
        type = ...
    }
}
~~~

...and not

~~~lua
{
    type = "container",
    x = 1,
    y = 1,
    {
        {type = ...}
    }
}
~~~
~~~